spring-cql/src/test/java/org/springframework/cassandra/test/integration/EmbeddedCassandraServerHelper.java

```diff
@@ -0,0 +1,357 @@
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.db.commitlog.CommitLog;
+import org.apache.cassandra.io.util.FileUtils;
+import org.apache.cassandra.service.CassandraDaemon;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.SocketUtils;
+import org.yaml.snakeyaml.reader.UnicodeReader;
+
+/**
+ * Imported Embedded Cassandra server startup helper.
+ *
+ * @author Mark Paluch
+ */
+class EmbeddedCassandraServerHelper {
+
+	private static Logger log = LoggerFactory.getLogger(EmbeddedCassandraServerHelper.class);
+
+	public static final long DEFAULT_STARTUP_TIMEOUT = 10000;
+	public static final String DEFAULT_TMP_DIR = "target/embeddedCassandra";
+	/** Default configuration file. Starts embedded cassandra under the well known ports */
+	private static final String INTERNAL_CASSANDRA_KEYSPACE = "system";
+	private static final String INTERNAL_CASSANDRA_AUTH_KEYSPACE = "system_auth";
+	private static final String INTERNAL_CASSANDRA_TRACES_KEYSPACE = "system_traces";
+
+	private static CassandraDaemon cassandraDaemon = null;
+	private static String launchedYamlFile;
+
+	public static void startEmbeddedCassandra(String yamlFile, long timeout) throws Exception {
+		startEmbeddedCassandra(yamlFile, DEFAULT_TMP_DIR, timeout);
+	}
+
+	public static void startEmbeddedCassandra(String yamlFile, String tmpDir, long timeout) throws Exception {
+		if (cassandraDaemon != null) {
+			/* nothing to do Cassandra is already started */
+			return;
+		}
+
+		if (!StringUtils.startsWith(yamlFile, "/")) {
+			yamlFile = "/" + yamlFile;
+		}
+
+		rmdir(tmpDir);
+		copy(yamlFile, tmpDir);
+		File file = new File(tmpDir + yamlFile);
+		readAndAdaptYaml(file);
+		startEmbeddedCassandra(file, tmpDir, timeout);
+	}
+
+	/**
+	 * Set embedded cassandra up and spawn it in a new thread.
+	 */
+	public static void startEmbeddedCassandra(File file, String tmpDir, long timeout) throws Exception {
+		if (cassandraDaemon != null) {
+			/* nothing to do Cassandra is already started */
+			return;
+		}
+
+		checkConfigNameForRestart(file.getAbsolutePath());
+
+		log.debug("Starting cassandra...");
+		log.debug("Initialization needed");
+
+		System.setProperty("cassandra.config", "file:" + file.getAbsolutePath());
+		System.setProperty("cassandra-foreground", "true");
+		System.setProperty("cassandra.native.epoll.enabled", "false"); // JNA doesnt cope with relocated netty
+
+		cleanupAndLeaveDirs();
+		final CountDownLatch startupLatch = new CountDownLatch(1);
+		ExecutorService executor = Executors.newSingleThreadExecutor();
+		executor.execute(new Runnable() {
```

Comment:
Maybe consider the use of `submit(:Runnable)` returning a `Future`.  This will simplify the following code by eliminating the explicit need for a `CountDownLatch`.